server/src/lfortran-language-server.ts

```diff
@@ -320,10 +320,16 @@ export class LFortranLanguageServer {
     this.settings = await this.getDocumentSettings(uri);
     this.logger.configure(this.settings);
     const document = this.documents.get(uri);
-    const text = document?.getText();
-    if (typeof text === "string") {
+    if (document !== undefined) {
+      const text = document.getText();
       const line = request.position.line;
-      const column = request.position.character;
+      let column = request.position.character;
+      if (column > 0) {
+        const offset: number = document.offsetAt(request.position);
+        if ((offset == text.length) || !RE_IDENTIFIABLE.test(text[offset])) {
+          column--;  // might be at right side of word boundary
+        }
+      }
```

Comment:
Is this something that `lfortran-lsp` should do, or is this something that LFortran itself should fix?

Reply:
We can fix it in the compiler. To me, it seems like we should consider whether the cursor is literally over one of a symbol's characters before jumping to its definition. I thought it might be a VSCode quirk that symbols are highlighted when the cursor is to the right of their word boundaries but it seems the fortran language server that emacs uses will jump to the definition of a symbol when the cursor is to the right of its word boundary, as well. If this is the behavior that users expect then we should adopt it into the compiler so we do not have to duplicate this logic in every client.

Reply:
That's what I was thinking. It seems the logic is that if you put the cursor on the space right after a word, it will get highlighted. But if there are more spaces and you move to the right, it won't get highlighted. So we can implement a similar logic in LFortran.